File: src/infrastructure/messagesRepository.js
var message = require('../domain/core/message');

var MessageRepository = function MessagesRepository(eventsStore){
    var getAllEvents = function getAllEvents(messageId){
        return eventsStore.getEventsOfAggregate(messageId);
    };

    this.getMessage = function getMessage(messageId){
        var events = getAllEvents(messageId);
        return message.create(events);
    };

    this.getDescription = function getDescription(messageId){
        var events = getAllEvents(messageId);
        var messageQuacked = events[0];

        return new message.MessageDescription(messageQuacked.author, messageQuacked.content);
    };
};

exports.create = function create(eventsStore){
    return new MessageRepository(eventsStore);
};

File: Mixter.Infrastructure/MessagesRepository.cs
using System.Linq;
using Mixter.Domain.Core.Messages;
using Mixter.Domain.Core.Messages.Events;

namespace Mixter.Infrastructure
{
    public class MessagesRepository
    {
        private readonly EventsStore _eventsStore;

        public MessagesRepository(EventsStore eventsStore)
        {
            _eventsStore = eventsStore;
        }

        public Message Get(MessageId id)
        {
            return new Message(_eventsStore.GetEventsOfAggregate(id));
        }

        public MessageDescription GetDescription(MessageId id)
        {
            var creationEvent = _eventsStore.GetEventsOfAggregate(id).OfType<MessageQuacked>().First();

            return new MessageDescription(creationEvent);
        }
    }
}
